relational_operators/DestroyHashOperator.hpp

```diff
@@ -75,20 +89,28 @@ class DestroyHashWorkOrder : public WorkOrder {
   /**
    * @brief Constructor.
    *
-   * @param hash_table_index The index of the JoinHashTable in QueryContext.
+   * @param hash_table_group_index The group index of the JoinHashTable in QueryContext.
    * @param query_context The QueryContext to use.
    **/
-  DestroyHashWorkOrder(const QueryContext::join_hash_table_id hash_table_index,
-                       QueryContext *query_context)
-      : hash_table_index_(hash_table_index),
-        query_context_(DCHECK_NOTNULL(query_context)) {}
+  DestroyHashWorkOrder(const CatalogRelation &input_relation,
```

Comment:
@adalbertgerald We could not use `CatalogRelation` in any `WorkOrder`s. To get the partition info, you need to put it in `WorkOrder` proto.